example-model/src/test/java/com/appunite/rx/example/model/presenter/CreatePostPresenterTest.java

```diff
@@ -0,0 +1,267 @@
+
+    @Test
+    public void testAfterNameIsNullSendAndSubmitClick_showNameEmptyError() throws Exception {
+        final TestSubscriber<Object> nameEmptyError = new TestSubscriber<>();
+        postPresenter.showNameIsEmptyErrorObservable().subscribe(nameEmptyError);
+        postPresenter.nameObservable().onNext(null);
+        postPresenter.bodyObservable().onNext("pies");
+
+        postPresenter.sendObservable().onNext(null);
+
+        assert_().that(nameEmptyError.getOnNextEvents()).hasSize(1);
+    }
+
+    @Test
+    public void testAfterBodyIsEmptyStringSendAndSubmitClick_showBodyEmptyError() throws Exception {
+        final TestSubscriber<Object> bodyEmptyError = new TestSubscriber<>();
+        postPresenter.showBodyIsEmptyErrorObservable().subscribe(bodyEmptyError);
+        postPresenter.nameObservable().onNext("pies");
+        postPresenter.bodyObservable().onNext("");
+
+        postPresenter.sendObservable().onNext(null);
+
+        assert_().that(bodyEmptyError.getOnNextEvents()).hasSize(1);
+    }
+
+    @Test
+    public void testAfterNameIsEmptyStringSendAndSubmitClick_showNameEmptyError() throws Exception {
+        final TestSubscriber<Object> nameEmptyError = new TestSubscriber<>();
+        postPresenter.showNameIsEmptyErrorObservable().subscribe(nameEmptyError);
+        postPresenter.nameObservable().onNext("");
+        postPresenter.bodyObservable().onNext("pies");
+
+        postPresenter.sendObservable().onNext(null);
+
+        assert_().that(nameEmptyError.getOnNextEvents()).hasSize(1);
+    }
+
+    @Test
+    public void testAfterStart_progressBarIsHidden() throws Exception {
+        final TestSubscriber<Boolean> showProgress = new TestSubscriber<>();
+        postPresenter.progressObservable().subscribe(showProgress);
+
+        assert_().that(showProgress.getOnNextEvents()).isEqualTo(ImmutableList.of(false));
+    }
+
+    @Test
+    public void testAfterFillingDataAndClickSend_progressBarIsShowed() throws Exception {
+        final TestSubscriber<Object> showProgress = new TestSubscriber<>();
+        postPresenter.progressObservable().subscribe(showProgress);
+        fillDataAndSubmit();
+
+        assert_().that(showProgress.getOnNextEvents()).isEqualTo(ImmutableList.of(false, true));
+    }
+
+    @Test
+    public void testAfterSaveSucces_progressBarIsHidden() throws Exception {
+        final TestSubscriber<Object> hideProgressBar = new TestSubscriber<>();
+        postPresenter.progressObservable().subscribe(hideProgressBar);
+        fillDataAndSubmit();
+        returnCorrectResponse();
+
+        assert_().that(hideProgressBar.getOnNextEvents()).isEqualTo(ImmutableList.of(false,true,false));
+    }
+
+    @Test
+    public void testAfterSaveFail_progressBarIsHidden() throws Exception {
+        final TestSubscriber<Object> hideProgressBar = new TestSubscriber<>();
+        postPresenter.progressObservable().subscribe(hideProgressBar);
+        fillDataAndSubmit();
+        returnException();
```

Comment:
should be enter before this line, remember:
* setup
<line brake>
* action
<line brake>
* check